### ISSUE TITLE
Questions

### ISSUE THREAD

**Issue by Swoorup** [insanitybit/dgraph-rs#1]
@insanitybit I see your comments on https://github.com/davidB/dgraph_client-rs/issues/8

And probably guessed, this covers additional features (transactions). 

I am willing to take upon maintaining it. Any ideas what needs to be done to run an example and pass the tests mentioned here it such that it becomes an official rust client? (https://discuss.dgraph.io/t/dgraph-client-in-rust/2189/2)

Also did you face any issues with dgraph on its own?

**Comment by insanitybit** [comment]
Hi Swoorup,

I'd rather not hand over maintainership in its entirety, but I'd be happy to see PRs, especially around testing. I intend to maintain this fairly actively, so I don't think transfer of ownership is necessary.

I'm using this library right now for transactions, and it works for my use cases, but I'd love to see it extended.

As for integration tests, the Go code is fairly simple. My dgraph client is basically a 1 to 1 port, and I'd imagine that a 1 to 1 port of the integration tests will be similarly easy to do.

I'd say:
* Get a local DGraph server running
* Port the tests over to the Rust client
* Write a 'harness' to get dgraph's server started, and then orchestrate the tests (basically a trivial shell script would work)
* Get the tests over to something like travis-ci so that we can automate the tests

I would welcome PRs that provide any part of any of those steps.

I haven't had any issues with DGraph itself - it runs very well for me, and I've written integration tests with it in the past. 

Thanks for letting me know that this is something you're interested in - I didn't know that anyone was using DGraph from Rust, so I haven't put work into this library. I can definitely change that.

**Comment by Swoorup** [comment]
Thanks for the reply. I still prefer pingcap's grpc over the rust native grpc as its more feature complete and more performant for my use case. 

And I think Transactions api needs to look more like python and javascript ones.

https://github.com/dgraph-io/pydgraph
https://github.com/dgraph-io/dgraph-js#create-a-transaction

What are your thoughts?

**Comment by insanitybit** [comment]
I agree. I hadn't used pydgraph until recently, and I think sticking to a similar API makes sense.

Which grpc crate are you referring to?

**Comment by Swoorup** [comment]
grpcio by pingcap. I will probably do a complete rewrite since it requires so.

**Comment by insanitybit** [comment]
Awesome - yeah, if you end up rewriting this I'd be happy to see it. I built this library to satisfy a minimal requirement for some other code I write. I'd love to see it get closer to the pydgraph API, but I have just literally no free time right now.

**Comment by Swoorup** [comment]
https://github.com/Swoorup/dgraph-rs

Just missing jwt login implementation

**Comment by insanitybit** [comment]
My understanding was that the jwt login stuff was beta? But maybe that's changed. Anyways - looks great, I'll probably switch over to that library and deprecate this crate sometime soon.
